bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/hbasesnapshots/ImportJobFromHbaseSnapshot.java

```diff
@@ -100,6 +111,19 @@ public static void main(String[] args) throws Exception {
   @VisibleForTesting
   static Pipeline buildPipeline(ImportOptions opts) throws Exception {
 
+    if(opts.getEnableSnappy() != null && opts.getEnableSnappy()) {
+        DataflowPipelineOptions dataFlowOpts = opts.as(DataflowPipelineOptions.class);
+        dataFlowOpts.setSdkContainerImage(CONTAINER_IMAGE_PATH_PREFIX + ReleaseInfo.getReleaseInfo().getVersion());
+        List<String> expOpts = dataFlowOpts.getExperiments();
+        if (expOpts != null) {
+            expOpts.add("use_runner_v2");
+            dataFlowOpts.setExperiments(expOpts);
+        } else {
+            final List<String> exps = new ArrayList<String>(Arrays.asList("use_runner_v2"));
+            dataFlowOpts.setExperiments(exps);
+        }
+    }
```

Comment:
You should avoid mutating lists inplace. Also, you should not add the flag if its already there:

```java
List<String> expOpts = Objects.firstNonNull(dataFlowOpts.getExperiments(), new ArrayList());
if (!expOpts.contains("use_runner_v2")) {
  expOpts = new ArrayList<>(existingOpts);
  expOpts.add("use_runner_v2")
}
dataflowOpts.setExperiments(expOpts);
```